src/runtime/lib/API.ts

```diff
@@ -228,3 +231,18 @@ export abstract class AbstractWebView {
         );
     }
 }
+
+export abstract class AbstractTreeView {
+    protected abstract view: TreeView<PluginTreeItem<Resource>>;
+    public context: ExtensionContext;
```

Comment:
Can we streamline this across all the abstractions? I guess calling it `extensionContext` makes it more specific that it stores extension related context rather than the component related context.

Reply:
good call, will rename it to extensionContext